### ISSUE TITLE
file written simultaneously by different rules

### ISSUE THREAD

**Issue by singerj** [snakemake/snakemake#2070]
Many thanks for this awesome tool. It makes a lot of things so much easier. However, I think the following is a bug and not a feature.

**Snakemake version**
7.20.0

**Describe the bug**
If _ruleorder_ is used the same file can be written by different rules simultaneously.

**Minimal example**
```
rule all:
    input:
        "a.txt",
        "b.txt",
        "c.txt"

ruleorder: a_b > b_c
rule a_b:
    output:
        "a.txt",
        "b.txt"

rule b_c:
    output:
        "b.txt",
        "c.txt"
```

**Additional context**
Without the _ruleorder_ snakemake raises an AmbiguousRuleException.

**Comment by pvandyken** [comment]
> Without the ruleorder snakemake raises an AmbiguousRuleException.

This is correct and intentional. Otherwise, Snakemake won't know which rule to choose.

I don't think I understand the behaviour you're looking for... perhaps you have a more complete example?

**Comment by pvandyken** [comment]
Just to be clear, you're saying that given the `Snakefile` you have there, both `rule a_b` and `rule b_c` will run, causing `b.txt` to be written twice?

Are you hoping to get this minimal example to work in some sensible way? Or were you expecting this to error out?

**Comment by dariober** [comment]
> Are you hoping to get this minimal example to work in some sensible way? Or were you expecting this to error out?

My 2p: I think the OP's example should throw an error. Rules `a_b` and `b_c` can run in parallel which means you don't know which rule produced `b.txt`. In fact, you cannot even know whether `b.txt` is sane since it could be a mixture of the two rules writing to the same file

**Comment by singerj** [comment]
Sorry for not being clear. Just as @dariober pointed out, the status of ```b.txt``` will be unclear. Therefore, I would have expected Snakemake to error out with an AmbiguousRuleException, just as in the case when no rule order is given.

**Comment by pvandyken** [comment]
Gotcha, I agree that should be the behaviour

**Comment by SichongP** [comment]
I don't know if this was intended at all but *if* rule order was specified, doesn't that mean user is aware of and has addressed this conflict?
